cypress/e2e/1-todo/todo.cy.js

```diff
@@ -0,0 +1,183 @@
+import 'cypress-real-events/support';
+
+describe('todo list app', () => {
+  it('할 일이 존재하지 않는다면 "There are no to-do items. Please write your to-dos" 가 나타난다.', () => {
+    cy.visit('http://localhost:5173');
+    cy.get('.todo-list-item').should('not.exist');
+    cy.get('.todo-empty-msg').should(
+      'contain',
+      'There are no to-do items. Please write your to-dos'
+    );
+  });
+
+  describe('할 일을 누르고 엔터를 누르면', () => {
+    it('목록에 노출된다', () => {
+      cy.visit('http://localhost:5173');
+      cy.get('.todo-input').type('밥먹기{enter}');
+      cy.get('.todo-list-item').should('contain', '밥먹기');
+    });
+
+    it('완료되지않은 갯수가 +1 된다', () => {
+      cy.visit('http://localhost:5173');
+      cy.get('.todo-input').type('밥먹기{enter}');
+      cy.get('#incomplete').should('contain', 1);
+    });
+  });
+
+  describe('할 일 항목을 체크하면', async () => {
+    it('완료된 항목은 목록 하단으로 이동하고, 완료된 항목의 갯수가 갱신된다', () => {
+      cy.visit('http://localhost:5173');
+      cy.get('.todo-input').type('밥먹기1{enter}');
+      cy.get('.todo-input').type('밥먹기2{enter}');
+      cy.get('.todo-input').type('밥먹기3{enter}');
+
+      cy.get('.todo-list-item:first-child input').should('not.be.checked');
+      cy.get('.todo-list-item:first-child input')
+        .invoke('attr', 'id')
+        .then((beforeKey) => {
+          cy.get('.todo-list-item:first-child input').click();
+
+          cy.get('.todo-list-item:last-child input')
+            .invoke('attr', 'id')
+            .then((afterKey) => {
+              expect(beforeKey).to.equal(afterKey);
+            });
+        });
+
+      cy.get('#complete').should('contain', 1);
+    });
+
+    it('완료된 항목을 체크해제하면 목록 상단으로 이동하고, 미완료된 항목의 갯수가 갱신된다', () => {
+      cy.visit('http://localhost:5173');
+      cy.get('.todo-input').type('밥먹기1{enter}');
+      cy.get('.todo-input').type('밥먹기2{enter}');
+      cy.get('.todo-input').type('밥먹기3{enter}');
+      cy.get('.todo-list-item:first-child input').click();
+
```

Comment:
p5. 이 부분도 다른 곳처럼 하나의 describe에 대해 사전 행위가 중복되는 거는 beforeEach로 분리하면 중복 코드를 줄일 수 있을 것 같아요! `cy.visit('http://localhost:5173');` 이 부분은 위 코멘트에 남긴대로 분리했다고 가정하고 제안 코드에서 빼둔 거라 참고 부탁드려요

```suggestion
describe('할 일 항목을 체크하면', async () => {
    beforeEach(() => {
      cy.get('.todo-input').type('밥먹기1{enter}');
      cy.get('.todo-input').type('밥먹기2{enter}');
      cy.get('.todo-input').type('밥먹기3{enter}');
    });

    it('완료된 항목은 목록 하단으로 이동하고, 완료된 항목의 갯수가 갱신된다', () => {
      cy.get('.todo-list-item:first-child input').should('not.be.checked');
      cy.get('.todo-list-item:first-child input')
        .invoke('attr', 'id')
        .then((beforeKey) => {
          cy.get('.todo-list-item:first-child input').click();

          cy.get('.todo-list-item:last-child input')
            .invoke('attr', 'id')
            .then((afterKey) => {
              expect(beforeKey).to.equal(afterKey);
            });
        });

      cy.get('#complete').should('contain', 1);
    });

    it('완료된 항목을 체크해제하면 목록 상단으로 이동하고, 미완료된 항목의 갯수가 갱신된다', () => {
      cy.get('.todo-list-item:first-child input').click();

```